boards/tests/test_template.py

```diff
@@ -30,3 +31,43 @@ def test_home_url_with_empty_board_list(self):
         Board.objects.all().delete()
         response = self.client.get(reverse("home"))
         self.assertEquals(0, len(response.context.get("boards")))
+
+
+class TopicListViewTests(TestCase):
+    """It tests template list of Topics/url named all_topics"""
+
+    def setUp(self):
+        topic_objects_list = []
+        board_associated_with_topic = Board.objects.create(
+            name="board", slug="boardss", description="1"
+        )
+        user_associated_with_topic = User.objects.create_user(
+            username="sourabh", email="so@dj.com", password="Somepassword"
+        )
+
+        for i in range(10):
+            topic_objects_list.append(
+                Topic(
+                    board=board_associated_with_topic,
+                    slug=str(i),
+                    user=user_associated_with_topic,
+                    subject=str(i),
+                )
+            )
+        Topic.objects.bulk_create(objs=topic_objects_list)
+
+    def tearDown(self):
+        Topic.objects.all().delete()
+
+    def test_rendering_of_all_topics_url_provides_correct_template(self):
+        response = self.client.get(reverse("all_topics"))
+        self.assertTemplateUsed(response, "boards/topic_list.html")
```

Comment:
Added "topic_" in filenames to avoid name conflict as there was already list.html file.

Reply:
Have your templates at the project level instead of app-level because when the apps grow it will be difficult to maintain the templates within the app directory. Take a look at https://github.com/pydanny/cookiecutter-django is a template for bootstrapping a Django project with a standard structure we can borrow some tips from it in organizing since we won't be needing all of it.

You can ignore this in this PR and make these from django cookiecutter and push it directly to master.